dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkAspect.java

```diff
@@ -173,7 +177,12 @@ private void commitTransaction(UnitOfWork unitOfWork, Session session) {
     private static void setContext(UnitOfWork unitOfWork, Session session) {
         ManagedSessionContext.bind(session);
         UnitOfWorkContext.setUnitOfWork(unitOfWork);
-        UnitOfWorkContext.setSessionFactory(session.getSessionFactory());
+        SessionFactory sessionFactory = session.getSessionFactory();
+        if (sessionFactory == null) {
```

Comment:
Assuming i'm wrong about the above, this should be a good failsafe. This is the only spot that sets the value we fail to get from the getter in our errors.